### ISSUE TITLE
Can't connect to kernel extension

### ISSUE THREAD

**Issue by perlun** [tekezo/Karabiner#393]
Hi,

I get the error "Karabiner canot connect with kernel extension. Please restart" issue rather often. It's quite annoying. Anything I can do about it? Running latest released Karabiner version.

OSX version 10.10.3 (14D113c).

**Comment by tekezo** [comment]
Is there any log message?
You can confirm them by Console.app.
https://pqrs.org/osx/karabiner/faq.html.en#general

**Comment by perlun** [comment]
Something like this. Does it help?

```
07/05/15 09:01:02,135 Karabiner[485]: [ERROR] BRIDGE_USERCLIENT_OPEN returned 0xe00002c5
07/05/15 09:01:02,135 Karabiner[485]: [INFO] IOService is not found.
07/05/15 09:01:02,000 kernel[0]: org.pqrs.Karabiner --Error-- UserClient_kext::callback_open kIOReturnExclusiveAccess
07/05/15 09:01:02,635 Karabiner[485]: [ERROR] BRIDGE_USERCLIENT_OPEN returned 0xe00002c5
07/05/15 09:01:02,635 Karabiner[485]: [INFO] IOService is not found.
07/05/15 09:01:03,000 kernel[0]: org.pqrs.Karabiner --Error-- UserClient_kext::callback_open kIOReturnExclusiveAccess
07/05/15 09:01:03,135 Karabiner[485]: [ERROR] BRIDGE_USERCLIENT_OPEN returned 0xe00002c5
07/05/15 09:01:03,135 Karabiner[485]: [INFO] IOService is not found.
07/05/15 09:01:56,495 Karabiner[485]: [INFO] BRIDGE_USERCLIENT_SYNCHRONIZED_COMMUNICATION connection is null
07/05/15 09:02:03,048 Karabiner[485]: [INFO] BRIDGE_USERCLIENT_SYNCHRONIZED_COMMUNICATION connection is null
07/05/15 09:02:03,050 Karabiner[485]: [INFO] BRIDGE_USERCLIENT_SYNCHRONIZED_COMMUNICATION connection is null
07/05/15 09:02:05,222 Karabiner[485]: [INFO] BRIDGE_USERCLIENT_SYNCHRONIZED_COMMUNICATION connection is null
07/05/15 09:02:07,588 Karabiner[485]: [INFO] BRIDGE_USERCLIENT_SYNCHRONIZED_COMMUNICATION connection is null
```

**Comment by tekezo** [comment]
Did you upgrade Karabiner while multiple users are logged in?

Please restart your mac if you can.

If not, please execute folloing commands in Terminal.
These commands terminate all Karabiner processes and unload old kernel extension.
```
sudo killall Karabiner
/Applications/Karabiner.app/Contents/Library/bin/kextload unload   
```
Then open Karabiner.

**Comment by perlun** [comment]
I have multiple users logged on, but I don't think I've upgraded now for a while. But yes, it's very likely that this happens after multiple users have been using Karabiner (switching between them while they are still logged on). Is that an unsupported scenario or should it work?

(Thanks for the hint - rebooted now already, but will try that for next time.)

**Comment by tekezo** [comment]
Thank you for information!

I was remembered that OS X has a bug that user suspend process at user switching sometimes does not work properly.
Therefore, previous user keeps grabbing exclusive resource and new user cannot use Karabiner.

I've added a workaround for this OS X issue at the latest beta version (v10.6.51).
Please confirm it.

How to upgrade to the latest beta version:
https://pqrs.org/osx/karabiner/document.html#check-for-updates

**Comment by perlun** [comment]
Thanks @tekezo. I have the beta installed now, it managed to kill the existing instance now at least (I got the error now when switching from one user to another - I use a "private" account for my hobby activities and a "work" account for work, so I switched to the latter now when getting back into the office).

Will let you know if it *doesn't* help, so if you don't hear from me again, you can assume this is closed within a few weeks or so. Thanks a lot!

**Comment by perlun** [comment]
Can't see anything indicating that it causes problems. Any idea about when the beta can be released?

**Comment by tekezo** [comment]
I was testing the Karabiner several weeks for release.
And now, I released Karabiner v10.7.0. Enjoy!
https://pqrs.org/osx/karabiner/
